src/glyph/parse.rs

```diff
@@ -123,18 +123,66 @@ impl<'names> GlifParser<'names> {
             }
         }
 
+        // UFO contour sanity constraints:
+        // 1. If a "move" occurs, it must be the first point of a contour. ufoLib
+        //    may allow this for format 1, we don't.
+        // 2. The "smooth" attribute must not be set on off-curve points.
+        // 3. An off-curve point must be followed by a curve or qcurve.
+        // 4. A maximum of two offcurves can precede a curve.
         let mut points = Vec::new();
+        let mut number_of_offcurves: u32 = 0;
+        let mut contour_closed: bool = true;
         loop {
             match reader.read_event(buf)? {
                 Event::End(ref end) if end.name() == b"contour" => break,
                 Event::Empty(ref start) if start.name() == b"point" => {
                     let point = self.parse_point(reader, start)?;
+                    match point.typ {
+                        PointType::Move => {
+                            if !points.is_empty() {
+                                return Err(err!(reader, ErrorKind::UnexpectedMove));
+                            }
+                            contour_closed = false;
+                        }
```

Comment:
I would include the condition in the match arm.
```suggestion
                        PointType::Move if !points.is_empty() => return Err(err!(reader, ErrorKind::UnexpectedMove)),
```

Reply:
I'll try to use the Pen thing instead! Maybe that works.